### PR TITLE
Fix take and make a test

### LINKED ISSUE #851: `cupy.take` bug when `axis` is not `None`
```
>>> x=numpy.arange(12).reshape((3,4))
>>> x.take([1,2],axis=1)
array([[ 1,  2],
       [ 5,  6],
       [ 9, 10]])
>>> y=cupy.arange(12).reshape((3,4))
>>> y.take([1,2],axis=1)
array([[1, 2],
       [3, 4],
       [5, 6]])
```

### PR DESCRIPTION
**Pull request by unnonouno**
fix #851

### CODE REVIEW

File: tests/cupy_tests/core_tests/test_ndarray.py [hunk 1]
```diff
@@ -45,3 +45,27 @@ def test_unsupported_type(self):
...
+@testing.parameterize(
+    {'shape': (3, 4, 5), 'inds': (2,), 'axis': None},
+    {'shape': (3, 4, 5), 'inds': (2,), 'axis': 0},
+    {'shape': (3, 4, 5), 'inds': (2,), 'axis': 1},
+    {'shape': (3, 4, 5), 'inds': (2,), 'axis': 2},
+    {'shape': (3, 4, 5), 'inds': (2, 3), 'axis': None},
+    {'shape': (3, 4, 5), 'inds': (2, 3), 'axis': 0},
+    {'shape': (3, 4, 5), 'inds': (2, 3), 'axis': 1},
+    {'shape': (3, 4, 5), 'inds': (2, 3), 'axis': 2},
+)
```
Comment (delta2323) [inline tests/cupy_tests/core_tests/test_ndarray.py]: How about using `testing.product`?

Comment (delta2323) [inline tests/cupy_tests/core_tests/test_ndarray.py]: Please add test cases where `inds` is single integer.

**Comment by delta2323** [comment]
Please add test cases in which `ValueError` is raised due to axis overrun.

File: tests/cupy_tests/core_tests/test_ndarray.py (hunk 1 above)
Comment (unnonouno) [inline tests/cupy_tests/core_tests/test_ndarray.py]: Single integer is not supported now. We need to make another issue.

**Comment by delta2323** [comment]
As #861 is another problem, we merge it this PR without fixing the problem.

**Comment by delta2323** [comment]
Please merge or rebase the master branch.

**Comment by beam2d** [comment]
LGTM!

**Pull request merged**